dawn/renderer/components/DNav.js

```diff
@@ -0,0 +1,147 @@
+import React from 'react';
+import {
+  Navbar,
+  ButtonToolbar,
+  ButtonGroup } from 'react-bootstrap';
+import { remote } from 'electron';
+import smalltalk from 'smalltalk';
+import ConfigBox from './ConfigBox';
+import UpdateBox from './UpdateBox';
+import StatusLabel from './StatusLabel';
+import TooltipButton from './TooltipButton';
+
+const storage = remote.require('electron-json-storage');
+
+class DNav extends React.Component {
+  static saveAddress(currentAddress) {
+    const prompt = smalltalk.prompt(
+      'Enter the IP address of the robot:',
+      'Examples: 192.168.13.100, 127.0.0.1',
+      currentAddress,
+    );
+    prompt.then((value) => {
+      if (/\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}/.test(value)) {
+        storage.set('runtimeAddress', {
+          address: value,
+        }, (err) => {
+          if (err) throw err;
+        });
+      } else if (value != null) {
+        console.log(`Bad IP: ${value}`);
+      }
+    }, () => console.log('Canceled'));
+  }
+
+  constructor(props) {
+    super(props);
+    this.toggleUpdateModal = this.toggleUpdateModal.bind(this);
+    this.toggleConfigModal = this.toggleConfigModal.bind(this);
+    this.state = {
+      showUpdateModal: false,
+      showConfigModal: false,
+    };
+  }
+
+  updateAddress() {
+    storage.has('runtimeAddress').then((hasKey) => {
+      if (hasKey) {
+        storage.get('runtimeAddress').then((data) => {
+          this.saveAddress(data.address);
+        });
+      } else {
+        this.saveAddress('192.168.13.100');
```

Comment:
Not sure what relevance this specific IP address has, but it should probably be moved to a `const` at the top of the file. (Also, there's another default in ConfigBox's placeholder *and* one in Ansible as `DEFAULT_IP`... so more consolidation would be good?)